2022/day06/day06.py

```diff
@@ -1,8 +1,10 @@
 
+
 def is_unique(str: str):
     if len(str) == len(set(str)):
         return True
-        
+
+
 def subroutine(str: str, length: int):
     i = 0
     while i < len(str):
```

Comment:
This solution is nice in that it's quite straightforward to read and understand. One potential downside for performance is that its runtime is essentially `n` x `m` where `n` is the length of the whole input message and `m` is the length of the marker. That is, besides the first and last few items in the message, you'll be processing every letter 4 times for part 1 and 14 times for part 2. Can you use some data structure to only process each letter once?